src/item/linkItem/linkItem.ts

```diff
@@ -47,18 +48,28 @@ export const getLinkExtra = <U extends LinkItemExtra>(
   extra: U,
 ): U[typeof ItemType.LINK] => extra[ItemType.LINK];
 
-export const getLinkThumbnailUrl = <U extends LinkItemExtra>(
-  extra: U,
-): string | undefined => {
+/**
+ * Returns url for a link given its extra
+ * Prioritize icons for small size
+ * @param {LinkItemExtra} extra
+ * @param {ThumbnailSizeType} [size=ThumbnailSize.Medium] size
+ * @returns url if exists, null otherwise
+ */
+export const getLinkThumbnailUrl = (
+  extra: LinkItemExtra,
+  size: ThumbnailSizeType = ThumbnailSize.Medium,
+): string | null => {
   const { thumbnails, icons } = getLinkExtra(extra);
 
-  if (thumbnails && thumbnails.length > 0) {
-    return thumbnails[0];
-  }
-  if (icons && icons.length > 0) {
-    return icons[0];
+  const thumbnailUrl = thumbnails?.length ? thumbnails[0] : null;
+  const iconUrl = icons?.length ? icons[0] : null;
```

Comment:
I tested and it looks like accessing an element out of the array size just returns `undefined`. so if you are ok with undefined you can simplify the code with: 
```suggestion
  const thumbnailUrl = thumbnails[0];
  const iconUrl = icons[0];
```
Or even: 
```ts
  const [thumbnailUrl] = thumbnails;
  const [iconUrl] = icons;
```

The `??` operators should work the same.